### PR TITLE
Change to use rvtysh when calling the show commands

### PR DESCRIPTION
**Pull request by xumia**
#### What I did
Change to use rvtysh when calling the show commands

### CODE REVIEW

File: utilities_common/bgp_util.py [hunk 1]
```diff
@@ -153,8 +153,11 @@ def run_bgp_command(vtysh_cmd, bgp_namespace=multi_asic.DEFAULT_NAMESPACE):
     if bgp_namespace is not multi_asic.DEFAULT_NAMESPACE:
         bgp_instance_id = " -n {} ".format(multi_asic.get_asic_id_from_name(bgp_namespace))
 
-    cmd = 'sudo vtysh {} -c "{}"'.format(
-        bgp_instance_id, vtysh_cmd)
+    vtysh = 'vtysh'
+    if vtysh_cmd.startswith('show'):
+        vtysh = 'rvtysh'
```
Comment (qiluo-msft) [inline utilities_common/bgp_util.py:158]: >rvtysh [](http://example.com/codeflow?start=17&length=6)

Let's extract it into a string constant into `sonic-py-common`. The name is arguable. How about `vtysh_show`?

**Comment by qiluo-msft** [comment]
> def run_bgp_command(vtysh_cmd, bgp_namespace=multi_asic.DEFAULT_NAMESPACE):

It's more natural to create a new function `run_bgp_show_command("ip bgp summary")`.

---
Refers to: utilities_common/bgp_util.py:150 in 85c7ee6. [](commit_id = 85c7ee6444321715899a1149da29f0eb32c8442d, deletion_comment = False)

File: utilities_common/bgp_util.py (hunk 1 above)
Comment (xumia) [reply utilities_common/bgp_util.py]: Fixed, changed to use the variable name vtysh_shell_cmd, and added the constants.

**Comment by xumia** [comment]
> > def run_bgp_command(vtysh_cmd, bgp_namespace=multi_asic.DEFAULT_NAMESPACE):
> 
> It's more natural to create a new function `run_bgp_show_command("ip bgp summary")`.
> 
> Refers to: utilities_common/bgp_util.py:150 in [85c7ee6](https://github.com/Azure/sonic-utilities/commit/85c7ee6444321715899a1149da29f0eb32c8442d). [](commit_id = [85c7ee6](https://github.com/Azure/sonic-utilities/commit/85c7ee6444321715899a1149da29f0eb32c8442d), deletion_comment = False)

If adding a new function run_bgp_show_command, we need to fix multiple references.
Add the command only use for show bgp, the old one can be removed, it will become changing name from run_bgp_command to run_bgp_show_command.

File: show/bgp_frr_v4.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def summary(namespace, display):
 def neighbors(ipaddress, info_type):
     """Show IP (IPv4) BGP neighbors"""
 
-    command = 'sudo vtysh -c "show ip bgp neighbor'
+    command = 'sudo rvtysh -c "show ip bgp neighbor'
```
Comment (qiluo-msft) [inline show/bgp_frr_v4.py:38]: >rvtysh [](http://example.com/codeflow?start=20&length=6)

Could you use the new constant `RVTYSH_COMMAND`? Or follow my other comment to define a new function `run_bgp_show_command` so you don't need to prefix with `sudo rvtysh -c`

Comment (xumia) [reply show/bgp_frr_v4.py:38]: @qiluo-msft , do you mean to change all the show commands to use the the new function?

Comment (qiluo-msft) [reply show/bgp_frr_v4.py:38]: yes

Comment (xumia) [reply show/bgp_frr_v4.py]: @qiluo-msft , could you please take a look again?

**Review by qiluo-msft — DISMISSED** [review]

**Comment by qiluo-msft** [comment]
Please solve the conflict.

File: tests/conftest.py [hunk 3]
```diff
@@ -128,14 +128,14 @@ def setup_single_bgp_instance(request):
         bgp_mocked_json = os.path.join(
             test_path, 'mock_tables', 'dummy.json')
 
-    def mock_show_bgp_summary(vtysh_cmd, bgp_namespace):
+    def mock_run_bgp_command(vtysh_cmd, bgp_namespace, vtysh_shell_cmd=''):
```
Comment (qiluo-msft) [inline tests/conftest.py:131]: >'' [](http://example.com/codeflow?start=71&length=2)

I think you should provide the same default value as the mocked function.

File: tests/conftest.py [hunk 4]
```diff
@@ -227,7 +228,7 @@ def mock_run_bgp_command_for_static(vtysh_cmd, bgp_namespace=""):
         else:
             return ""
 
-    def mock_run_bgp_command(vtysh_cmd, bgp_namespace):
+    def mock_run_bgp_command(vtysh_cmd, bgp_namespace, vtysh_shell_cmd=''):
```
Comment (qiluo-msft) [inline tests/conftest.py:231]: >vtysh_shell_cmd [](http://example.com/codeflow?start=55&length=15)

also this default parameter?

**Review by qiluo-msft — APPROVED** [review]

**Comment by yxieca** [comment]
@xumia do we still need this PR to be cherry-picked after #1648 is merged? This PR cannot be cherry-picked cleanly. Please create separate PR if some change is still needed.

**Pull request merged**